packages/@magic-sdk/react-native-bare/src/react-native-webview-controller.tsx

```diff
@@ -216,7 +229,18 @@ export class ReactNativeWebViewController extends ViewController {
   }
 
   protected async _post(data: any) {
+    const methodType = data.payload.method;
+    const messageId = data.payload.id;
+
     if (this.webView && (this.webView as any).postMessage) {
+      // Setup timeout for message response
+      const timeout = setTimeout(() => {
+        this.messageTimeouts.delete(messageId);
+        throw createResponseTimeoutError(`Response timed out for method: ${methodType} with message id: ${messageId}`);
+      }, 5000); // 5-second timeout
```

Comment:
Are we sure 5s is enough? We've had reports of calls to certain methods taking longer than 5s in regions like Asia and South Asia.

Reply:
Btw, do we know why the calls never return in some cases? I agree with having a time limit but if there's an issue on phantom that causes responses to hang we should probably look into that.

Reply:
@romin-halltari in this case, there could be multiple reasons for the delay. For the scope of this PR, the developer wants some indication that a response has been delayed, so we're focusing on just indicating that a delay has occurred as opposed to resolving any particular delay.

As for the 5 seconds, that was a random choice on my part. Given the realities of our users in Asia and South Asia I'll up it to 10 seconds. Thanks for the pointing that out!